src/SWP/Bundle/ContentBundle/EventListener/ProcessArticleMediaListener.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+/**
+ * This file is part of the Superdesk Web Publisher Content Bundle.
+ *
+ * Copyright 2016 Sourcefabric z.ú. and contributors.
+ *
+ * For the full copyright and license information, please see the
+ * AUTHORS and LICENSE files distributed with this source code.
+ *
+ * @copyright 2015 Sourcefabric z.ú.
+ * @license http://www.superdesk.org/license
+ */
+namespace SWP\Bundle\ContentBundle\EventListener;
+
+use Doctrine\Common\Persistence\ObjectManager;
+use Doctrine\ODM\PHPCR\Document\Generic;
+use SWP\Bundle\ContentBundle\Doctrine\ODM\PHPCR\ArticleMedia;
+use SWP\Bundle\ContentBundle\Doctrine\ODM\PHPCR\Image;
+use SWP\Bundle\ContentBundle\Doctrine\ODM\PHPCR\ImageRendition;
+use SWP\Bundle\ContentBundle\Event\ArticleEvent;
+use SWP\Bundle\ContentBundle\Manager\MediaManagerInterface;
+use SWP\Bundle\ContentBundle\Model\ArticleInterface;
+use SWP\Component\Bridge\Model\ItemInterface;
+use SWP\Component\Bridge\Model\Rendition;
+use SWP\Component\MultiTenancy\PathBuilder\TenantAwarePathBuilderInterface;
+
+class ProcessArticleMediaListener
+{
+    /**
+     * ProcessArticleMediaListener constructor.
+     *
+     * @param ObjectManager                   $objectManager
+     * @param TenantAwarePathBuilderInterface $pathBuilder
+     * @param string                          $mediaBasepath
+     * @param MediaManagerInterface           $mediaManager
+     */
+    public function __construct(
+        ObjectManager $objectManager,
+        TenantAwarePathBuilderInterface $pathBuilder,
+        $mediaBasepath,
+        MediaManagerInterface $mediaManager
+    ) {
+        $this->objectManager = $objectManager;
+        $this->pathBuilder = $pathBuilder;
+        $this->mediaBasepath = $mediaBasepath;
+        $this->mediaManager = $mediaManager;
+    }
+
+    /**
+     * @param ArticleEvent $event
+     */
+    public function onArticleCreate(ArticleEvent $event)
+    {
+        $package = $event->getPackage();
+        $article = $event->getArticle();
+        $this->objectManager->persist($article);
+
+        if (null !== $package && 0 === count($package->getItems())) {
+            return;
+        }
+
+        foreach ($package->getItems() as $key => $packageItem) {
+            // create document node for media
+            $mediaDocument = $this->createGenericDocument('media', $article);
+            if ($packageItem->getType() === 'picture' || $packageItem->getType() === 'file') {
+                $articleMedia = $this->handleMedia($article, $mediaDocument, $key, $packageItem);
+                $this->objectManager->persist($articleMedia);
+            }
+
+            if (0 !== count($packageItem->getItems())) {
```

Comment:
you can use `$packageItem->getItems()->count()` directly as it's type of ArrayCollection